### ISSUE TITLE
sails crash on broadcast when using redis and sockets 

### ISSUE THREAD

**Issue by OriTheMan** [balderdashy/sails#3035]
i try to use sockets with redis, but i receive error and the server crush when i try to broadcast data
i receive
```
(null):0
(null)

RangeError: Maximum call stack size exceeded
```

broadcast code:
```
sails.sockets.broadcast('Room-' + data.RoomId, 'Join', user, req.socket);
```

**Comment by OriTheMan** [comment]
complete code:

```
Users.create(data, function(err, user) {
					if(err) {
						return res.send(500);
					} else {
						//Find and return the users in room
						Users.find({...}).where({...}).exec(function(err, users) {
							if(err) {
								return res.send(500);
							} else {
								//Join the user to room
								sails.sockets.join(req.socket, 'Room-' + data.RoomId);
								sails.sockets.broadcast('Room-' + data.RoomId, 'Join', user, req.socket);
								return res.send(200,{Success: true, AllUsers: users, SomeData: some_data});	
							}
						});
					}
				});
```

when the ```sails.sockets.broadcast('Room-' + data.RoomId, 'Join', user, ``` executes, i receive the error, and there is no broadcasting

**Comment by sailsbot** [comment]
Thanks for posting, @OriTheMan.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
